### PR TITLE
2.4.0 stable release.

### PR DESCRIPTION
**Pull request by cijothomas**
Updated diagnostic source to 4.4.0
Base SDK to 2.4.0

DiagnosticSource adapter is still on 4.3 and 2.3 of diagsource and base sdk respectively. @pharring , @SergeyKanzhelev should i update diagsource adapter as well for this stable release?

### CODE REVIEW

File: GlobalStaticVersion.props [hunk 1]
```diff
@@ -8,12 +8,12 @@
...
     <SemanticVersionMinor>4</SemanticVersionMinor>
     <SemanticVersionPatch>0</SemanticVersionPatch>
-    <PreReleaseMilestone>beta2</PreReleaseMilestone>
+    <PreReleaseMilestone></PreReleaseMilestone>
     <!-- 
       Date when Semantic Version was changed. 
       Update for every public release.
     -->
-    <SemanticVersionDate>2017-05-07</SemanticVersionDate>
+    <SemanticVersionDate>2017-08-04</SemanticVersionDate>
```
Comment (SergeyKanzhelev) [inline GlobalStaticVersion.props]: do not change date when you are not changing major, minor or patch. It will break file versions

Comment (cijothomas) [inline GlobalStaticVersion.props]: ok. the comment says update for every public release?

**Comment by cijothomas** [comment]
Nuspecs take version automatically.

**Comment by cijothomas** [comment]
What about diagnostic source adapater? should it continue on 2.3 or update to 2.4?

**Comment by SergeyKanzhelev** [comment]
I'd say we need to upgrade to 2.4 to minimize support and promote upgrade to a new version. @pharring I remember you had concerns with some specific package. Was it diagnostics source adaptor?

File: src/Adapters/Log4NetAppender.Net45/Log4NetAppender.Net45.csproj [hunk 2]
```diff
@@ -30,12 +30,12 @@
       <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.ApplicationInsights, Version=2.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\..\packages\Microsoft.ApplicationInsights.2.4.0-beta3\lib\net45\Microsoft.ApplicationInsights.dll</HintPath>
+      <HintPath>..\..\..\..\packages\Microsoft.ApplicationInsights.2.4.0\lib\net45\Microsoft.ApplicationInsights.dll</HintPath>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />
     <Reference Include="System.Diagnostics.DiagnosticSource, Version=4.0.2.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
```
Comment (cijothomas) [inline src/Adapters/Log4NetAppender.Net45/Log4NetAppender.Net45.csproj]: These also need to be updated to 4.4.4

**Pull request merged**